pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -192,7 +195,7 @@ public void registerListener(Consumer<Notification> listener) {
                 });
 
                 return null;
-            }, executor);
+            }, listenerExecutor);
```

Comment:
I think this would introduce a lot of wired race conditions. Some of our current implementations relay on this single thread to keep data consistence and thread safe, like the `existsCache` and `childrenCache`.